lib/fathead/opam/parse.py

```
@@ -0,0 +1,151 @@
+
+        return contents
+
+
+class Document(object):
+    """
+    The Document class is responsible for setting up the end document
+    (output.txt). It is made up of 4 member functions and a dummy
+    constructor.
+    """
+    def __init__(self):
+        print("Document object instantiated")
+
+    def parse_contents(self, soup):
+        """
+        This member function parses the html document and extracts the
+        cells from the table.
+        """
+        data = []
+        table = soup.find('table', attrs={'id': 'packages'})
+        table_body = table.find('tbody')
+        rows = table_body.find_all('tr')
+
+        """
+        This section aims to extract the individual data points from the
+        table and turn them into a list. For each of the columns found in
+        <td> tags, we return them in a list, while finding one link.
+        We then preprocess the contents by stripping whitespace and
+        stripping the front of the relative link. Once this preprocessing
+        has been done, we connect the URL, and append it to the list.
+        """
+        for row in rows:
+            cols = row.find_all('td')
+            ref = row.find('a', href=True)
+
+            # Preprocessing
+            cols = [ele.text.strip() for ele in cols]
+            relurl = ref['href'].strip('../')
+
+            # Add to list
+            aburl = urljoin(__target__, relurl)
+            cols.append(aburl)
+            print(cols)
+            data.append(cols)
+
+        return data
+
+    def concat(self, name, description, url, version):
+        """
+        The concat (concatenation) member function is responsible for
+        preparing the data to be written to the file. The file is
+        layed outlike this as requested in the DuckDuckHack docs found
+        here: http://docs.duckduckhack.com/resources/fathead-overview.html#data-file-format
+        """
+        title = name or 'No title found'
```

This doesn't appear to be handled later? We should probably raise an exception or just return (then handle the 'undefined' value).